frame/support/procedural/src/lib.rs

```diff
@@ -818,18 +818,66 @@ pub fn config(_: TokenStream, _: TokenStream) -> TokenStream {
 /// The `#[pallet::constant]` attribute can be used to add an associated type trait bounded by `Get`
 /// from [`pallet::config`](`macro@config`) into metadata, e.g.:
 ///
-/// ```ignore
-/// #[pallet::config]
-/// pub trait Config: frame_system::Config {
-/// 	#[pallet::constant]
-/// 	type Foo: Get<u32>;
+/// ## Example:
+///
+/// ```
+/// #[frame_support::pallet]
+/// mod pallet {
+///     use frame_support::pallet_prelude::*;
+///
+///     # #[pallet::pallet]
+///     # pub struct Pallet<T>(_);
+///
+///     #[pallet::config]
+///     pub trait Config: frame_system::Config {
+/// 		/// This is like a normal `Get` trait, but it will be added into metadata.
+/// 		#[pallet::constant]
+/// 		type Foo: Get<u32>;
+/// 	}
 /// }
 /// ```
 #[proc_macro_attribute]
 pub fn constant(_: TokenStream, _: TokenStream) -> TokenStream {
 	pallet_macro_stub()
 }
 
+/// Declares an implementation block to be the dispatchable portion of a pallet. In other words,
+/// each function in the implementation block of this section is an extrinsic that can be called
+/// externally.
+///
+/// Other than than the `fn` attached to `Pallet`, this block will also generate an `enum Call`
+/// which encapsulates the different functions, alongside their arguments, except for `origin`.
+/// [`sp_runtime::traits::Dispatchable`] is then implemented for `enum Call`.
+///
+/// ## Example:
+///
+/// ```
+/// #[frame_support::pallet]
+/// mod pallet {
+///     use frame_support::pallet_prelude::*;
+///
+///     #[pallet::config]
+///     pub trait Config: frame_system::Config { }
+///
+///     #[pallet::pallet]
+///     pub struct Pallet<T>(_);
+///
+/// 	#[pallet::call]
+/// 	impl<T: Config> Pallet<T> {
+/// 		fn do_stuff(_origin: OriginFor<T>, arg: u32) {
```

Comment:
Is weight no longer required? I think that is a good thing, but was surprised there wasn't dev mode stuff going on

Reply:
It might be worth noting in the docs what `call` functions require? For example, it seems they all require an origin as the first argument.

Reply:
See #14115 for exactly what you said about `call`.